jmetal-algorithm/src/main/java/org/uma/jmetal/algorithm/multiobjective/moead/MOEADD.java

```diff
@@ -1478,22 +1470,19 @@ public int countTest() {
 
   @Override
   public String getName() {
-    return "MOEA/D-DD";
+        return "MOEADD";
   }
 
   @Override
   public String getDescription() {
-    return "MOEA/D-DD";
+        return "An Evolutionary Many-Objective Optimization Algorithm Based on Dominance and Decomposition";
   }
 
   public void replace(int position, S solution) {
     if (position > this.population.size()) {
       population.add(solution);
     } else {
       S toRemove = population.get(position);
-      rankSolution.remove(toRemove);
-      regionSolution.remove(toRemove);
-      associateDistSolution.remove(toRemove);
```

Comment:
Explicit removal becomes useless with weak maps, so these lines can be removed as well, although they can also stay there. I don't know whether explicit removal increases performance.